src/Proto.Cluster/SingleNode/SingleNodeActivatorActor.cs

```diff
@@ -0,0 +1,215 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="SingleNodeActivatorActor.cs" company="Asynkron AB">
+//      Copyright (C) 2015-2022 Asynkron AB All rights reserved
+// </copyright>
+// -----------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Proto.Cluster.SingleNode;
+
+class SingleNodeActivatorActor : IActor
+{
+    private static readonly ILogger Logger = Log.CreateLogger<SingleNodeActivatorActor>();
+
+
+    private readonly Cluster _cluster;
+    private readonly Dictionary<ClusterIdentity, PID> _actors = new();
+    private readonly HashSet<ClusterIdentity> _inFlightIdentityChecks = new();
+
+    public SingleNodeActivatorActor(Cluster cluster) => _cluster = cluster;
+
+    private Task OnStarted(IContext context)
+    {
+        var self = context.Self;
+        _cluster.System.EventStream.Subscribe<ActivationTerminated>(e => _cluster.System.Root.Send(self, e));
```

Comment:
Could this be simplified by using the `Subscribe<TMsg>(ISenderContext context, params PID[] pids)` overload?

Reply:
Yes, fixed :)